### PR TITLE
Add support for NSX-T Edge Gateway External Uplink configuration

### PR DESCRIPTION
**Pull request by Didainius**
This PR adds missing pieces to support NSX-T Edge Gateway External Network uplinks (only for external networks backed by NSX-T Segments) (VCD 10.4.1+)

List of things added:
* New internal function `*NsxtEdgeGateway.reorderUplinks()` that will order NSX-T Edge Gateway uplinks. There are two major types of uplinks
  * `NSXT_TIER0` or `NSXT_VRF_TIER0` - Tier 0 gateway backed external network (mandatory and is required during creation)
  * `IMPORTED_T_LOGICAL_SWITCH ` NSX-T segment backed external network (this is optional, attached post creation in "External networks" tab)
* New method `NsxtEdgeGateway.GetAllocatedIpCountByUplinkType` complementing existing
  `NsxtEdgeGateway.GetAllocatedIpCount`. It will return allocated IP counts by uplink types (requires VCD 10.4.1+)
* New method `NsxtEdgeGateway.GetPrimaryNetworkAllocatedIpCount` that will return total allocated IP
  count for primary uplink (T0 or T0 VRF)
* New field `types.EdgeGatewayUplinks.BackingType` that defines backing type of NSX-T Edge Gateway
  Uplink (available in VCD 10.4.1+)

_Note on uplink ordering_. API usually returns the mandatory network as element 0 in the resulting slice, but there have been cases where VCD sometimes shuffles these responses (don't have a proof for this particular endpoint). As a result - the `reorderUplinks` function is here to ensure that `NSXT_TIER0` or `NSXT_VRF_TIER0`uplink is always at position 0. It is also used in all functions related to NSX-T Edge Gateway.

Tested with tags `nsxt gateway network`:
* 10.5.0
* 10.4.1
* 10.4.0
* 10.4.2

Tests are for admin mode only, skip directive for Org tests is in place. The test requires 10.4.1+.

### CODE REVIEW

File: types/v56/types.go [hunk 1]
```diff
@@ -1619,18 +1619,19 @@ type DiskSection struct {
...
+	UnitNumber          int           `xml:"UnitNumber"`                // The device number on the SCSI or IDE controller of the disk.
+	BusNumber           int           `xml:"BusNumber"`                 //	The number of the SCSI or IDE controller itself.
+	AdapterType         string        `xml:"AdapterType"`               // The type of disk controller, e.g. IDE vs SCSI and if SCSI bus-logic vs LSI logic.
+	ThinProvisioned     *bool         `xml:"ThinProvisioned,omitempty"` // Specifies whether the disk storage is pre-allocated or allocated on demand.
+	Disk                *Reference    `xml:"Disk,omitempty"`            // Specifies reference to a named disk.
+	StorageProfile      *Reference    `xml:"StorageProfile,omitempty"`  // Specifies reference to a storage profile to be associated with the disk.
+	OverrideVmDefault   bool          `xml:"overrideVmDefault"`         // Specifies that the disk storage profile overrides the VM's default storage profile.
+	IopsAllocation      *IopsResource `xml:"IopsAllocation"`            // IOPS definition for the disk - added in 10.4 in replacement of 'iops'
+	Iops                *int64
```
Comment (Didainius) [inline types/v56/types.go:1632]: Temporary, until https://github.com/vmware/terraform-provider-vcd/pull/1108 is merged

File: govcd/nsxt_edgegateway_unit_test.go [hunk 2]
```diff
@@ -1640,3 +1640,119 @@ func TestOpenAPIEdgeGateway_DeallocateIpCount(t *testing.T) {
...
+			}},
+			want: args{edgeGatewayUplinks: []types.EdgeGatewayUplinks{
+				{BackingType: addrOf("NSXT_VRF_TIER0"), UplinkID: "5"},
+				{BackingType: addrOf("IMPORTED_T_LOGICAL_SWITCH"), UplinkID: "2"},
+				{BackingType: addrOf("IMPORTED_T_LOGICAL_SWITCH"), UplinkID: "3"},
+				{BackingType: addrOf("IMPORTED_T_LOGICAL_SWITCH"), UplinkID: "4"},
+				{BackingType: addrOf("IMPORTED_T_LOGICAL_SWITCH"), UplinkID: "1"},
+			}},
+		},
+		// Commented on purpose
```
Comment (lvirbalas) [inline govcd/nsxt_edgegateway_unit_test.go:1736]: Sounds mysterious! What is that purpose?

Comment (Didainius) [reply govcd/nsxt_edgegateway_unit_test.go]: It is an example failing test, that I used to validate that the testing mechanism itself works. Improved that comment.

File: govcd/api_vcd_test.go [hunk 3]
```diff
@@ -189,6 +189,7 @@ type TestConfig struct {
 			ExternalNetwork           string `yaml:"externalNetwork"`
 			EdgeGateway               string `yaml:"edgeGateway"`
 			NsxtImportSegment         string `yaml:"nsxtImportSegment"`
+			NsxtImportSegment2        string `yaml:"nsxtImportSegment2"`
```
Comment (dataclouder) [inline govcd/api_vcd_test.go:192]: Please add this field to `sample_govcd_test_config.yaml`

**Review by adambarreiro — APPROVED** [review]

File: govcd/api_vcd_test.go (hunk 3 above)
Comment (Didainius) [reply govcd/api_vcd_test.go:192]: Thanks!. I added to both now - govcd and Terraform as both were missing

**Review by lvirbalas — APPROVED** [review]
Thanks!

**Review by adezxc — APPROVED** [review]

**Review by dataclouder — APPROVED** [review]

**Pull request merged**